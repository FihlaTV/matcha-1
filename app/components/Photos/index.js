import React from 'react';
import { Text, View, StyleSheet, Image } from 'react-native';
import type { PhotoType } from 'app/types/Photo';
import { MARGINS, COLORS } from 'app/constants/design';
import Icon from 'react-native-vector-icons/FontAwesome';
import moment from 'moment';
import LoadingHOC from 'app/components/HOC/LoaderHOC';

const styles = StyleSheet.create({
  photoContainer: {
    backgroundColor: COLORS.BACKGROUND,
    borderColor: COLORS.GREY_LIGHTEST,
    borderWidth: 1,
    margin: MARGINS.SMALL,
    padding: MARGINS.SMALL,
  },
  photoControl: {
    flexDirection: 'row',
    marginBottom: MARGINS.SMALL,
  },
  photoHeader: {
    flexDirection: 'row',
    marginBottom: MARGINS.SMALL,
  },
  photoImage: {
    backgroundColor: COLORS.GREY_DARK,
    height: 325,
    margin: 0,
    marginBottom: MARGINS.SMALL,
    width: 325,
  },
  photoProfile: {
    backgroundColor: COLORS.GREY_DARK,
    borderRadius: 10,
    height: 20,
    marginRight: MARGINS.SMALL,
    width: 20,
  },
  photosContainer: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
  },
});

type PhotoProps = {
  photo: PhotoType,
};

const Photo = (props: PhotoProps) => (
  <View style={styles.photoContainer}>
    <View style={styles.photoHeader}>
      {/* eslint-disable-next-line */}
      <Image style={styles.photoProfile} source={{ uri: props.photo.user.picture }} defaultSource={require('app/assets/images/no-pic.jpg')} />
      <Text>{props.photo.user.username}</Text>
    </View>
    <Image style={styles.photoImage} source={{ uri: props.photo.url }} />
    <View style={styles.photoControl}>
      <Icon style={{ marginRight: MARGINS.SMALL }} name="heart" size={20} color={COLORS.GREY_DARK} />
      <Icon name="comment" size={20} color={COLORS.GREY_DARK} />
    </View>
    <Text>{moment(props.photo.createdAt).fromNow()}</Text>
  </View>
);

type PhotosProps = {
  photos: Array<PhotoType>,
};

const Photos = (props: PhotosProps) => (
  <View style={styles.photosContainer}>
    {
      props.photos.map((photo) => (<Photo key={photo.id} photo={photo} />))
    }
  </View>
);

export default LoadingHOC('photos')(Photos);
