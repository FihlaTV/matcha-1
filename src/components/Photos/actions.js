import type { Photo } from 'app/types/Photo';

export const LOAD_PHOTOS = '@PHOTOS_REDUCER/LOAD_PHOTOS';
export const ON_LOAD_PHOTOS = '@PHOTOS_SAGA/ON_LOAD_PHOTOS';

export const loadPhotos = (photos: Array<Photo>) => ({
  photos,
  type: LOAD_PHOTOS,
});

export const onLoadPhotos = () => ({
  type: ON_LOAD_PHOTOS,
});
