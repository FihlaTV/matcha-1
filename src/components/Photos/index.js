import React from 'react';
import { Text, View, StyleSheet, Image } from 'react-native';
import type { PhotoType } from 'app/types/Photo';
import { MARGINS, COLORS } from 'app/constants/design';
import Icon from 'react-native-vector-icons/FontAwesome';
import moment from 'moment';

const styles = StyleSheet.create({
  photoContainer: {
    backgroundColor: COLORS.BACKGROUND,
    borderColor: COLORS.GREY,
    borderWidth: 1,
    margin: MARGINS.SMALL,
    padding: MARGINS.SMALL,
  },
  photoControl: {
    flexDirection: 'row',
    marginBottom: MARGINS.SMALL,
  },
  photoHeader: {
    flexDirection: 'row',
    marginBottom: MARGINS.SMALL,
  },
  photoImage: {
    backgroundColor: COLORS.GREY_DARK,
    height: 325,
    margin: 0,
    marginBottom: MARGINS.SMALL,
    width: 325,
  },
  photoProfile: {
    backgroundColor: COLORS.GREY_DARK,
    borderRadius: 20,
    height: 20,
    marginRight: MARGINS.SMALL,
    width: 20,
  },
  photosContainer: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
  },
});

type PhotoProps = {
  photo: PhotoType,
};

const Photo = (props: PhotoProps) => (
  <View style={styles.photoContainer}>
    <View style={styles.photoHeader}>
      <Image style={styles.photoProfile} source={props.photo.user.picture} />
      <Text>{props.photo.user.username}</Text>
    </View>
    <Image style={styles.photoImage} source={props.photo.url} />
    <View style={styles.photoControl}>
      <Icon style={{ marginRight: MARGINS.SMALL }} name="heart" size={20} color={COLORS.GREY_DARK} />
      <Icon name="comment" size={20} color={COLORS.GREY_DARK} />
    </View>
    <Text>{moment(props.photo.createdAt).fromNow()}</Text>
  </View>
);

type PhotosProps = {
  photos: Array<PhotoType>,
};

const Photos = (props: PhotosProps) => (
  <View style={styles.photosContainer}>
    {
      props.photos.map((photo) => (<Photo key={photo.id} photo={photo} />))
    }
  </View>
);

export default Photos;
export { Photos };
