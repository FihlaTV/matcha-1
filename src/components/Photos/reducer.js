import { LOAD_PHOTOS } from 'app/components/Photos/actions';
import type { Photo } from 'app/types/Photo';

type State = {
  photos: Array<Photo>,
};

const initialState: State = {
  photos: [],
};

export default (state: State = initialState, action: any): State => {
  if (action.type === LOAD_PHOTOS) {
    return { ...state, photos: action.photos };
  }
  return state;
};
