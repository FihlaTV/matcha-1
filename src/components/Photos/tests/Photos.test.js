import React from 'react';
import Photos from 'src/components/Photos';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Photos', () => {
  describe('render test', () => {
    it('should render', () => {
      const wrapper = shallow(<Photos photos={[]} />).dive();
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
