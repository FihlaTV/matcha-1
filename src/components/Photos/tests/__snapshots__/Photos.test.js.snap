// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos render test should render 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-around",
    }
  }
/>
`;
